book/src/configuration.md

```diff
@@ -67,13 +67,19 @@ left = ["mode", "spinner"]
 center = ["file-name"]
 right = ["diagnostics", "selections", "position", "file-encoding", "file-line-ending", "file-type"]
 separator = "│"
+mode.normal= "NORMAL"
+mode.insert= "INSERT"
+mode.select= "SELECT"
 ```
 
 The following elements can be configured:
 
 | Key    | Description |
 | ------ | ----------- |
-| `mode` | The current editor mode (`NOR`/`INS`/`SEL`) |
+| `mode` | The current editor mode (`mode.normal`/`mode.insert`/`mode.select`) |
+| `mode.normal` | The text displayed by the `mode` element while in normal mode (defaults to `"NOR"`) |
+| `mode.insert` | The text displayed by the `mode` element while in insert mode (defaults to `"INS"`) |
+| `mode.select` | The text displayed by the `mode` element while in select mode (defaults to `"SEL"`) |
 | `spinner` | A progress spinner indicating LSP activity |
 | `file-name` | The path/name of the opened file |
```

Comment:
This table is a list of configurable elements you can put in the statusline's `left`, `center`, and `right` keys rather than the set of keys that are valid under `editor.statusline`. We should probably have a table for the keys that are valid and then a separate table for the elements.

Reply:
Something like this?

| Key    | Description |
| ------ | ----------- |
| `mode` | The current editor mode (`mode.normal`/`mode.insert`/`mode.select`) |
| `spinner` | A progress spinner indicating LSP activity |
| `file-name` | The path/name of the opened file |
| `file-encoding` | The encoding of the opened file if it differs from UTF-8 |
| `file-line-ending` | The file line endings (CRLF or LF) |
| `file-type` | The type of the opened file |
| `diagnostics` | The number of warnings and/or errors |
| `selections` | The number of active selections |
| `position` | The cursor position |
| `position-percentage` | The cursor position as a percentage of the total number of lines |
| `separator` | The string defined in `editor.statusline.separator` (defaults to `"│"`) |
| `spacer` | Inserts a space between elements (multiple/contiguous spacers may be specified) |

The statusline elements can be customized using the options below:
| Key    | Description |
| ------ | ----------- |
| `mode.normal` | The text displayed by the `mode` element while in normal mode (defaults to `"NOR"`) |
| `mode.insert` | The text displayed by the `mode` element while in insert mode (defaults to `"INS"`) |
| `mode.select` | The text displayed by the `mode` element while in select mode (defaults to `"SEL"`) |

Reply:
Along those lines, yeah. I was thinking

```markdown
The `[editor.statusline]` key takes the following sub-keys

| Key           | Description | Default |
| ---           | ---         | ---     |
| `left`        | A list of elements aligned to the left of the statusline | `["mode", "spinner", "file-name"]` |
| `center`      | A list of elements aligned to the middle of the statusline | `[]` |
| `right`       | A list of elements aligned to the right of the statusline | `["diagnostics", "selections", "position", "file-encoding"]` |
| `separator`   | The character used to separate elements in the statusline | `"|"` |
| `mode.normal` | The text shown in the `mode` element for normal mode | `"NOR"` |
| `mode.insert` | The text shown in the `mode` element for insert mode | `"INS"` |
| `mode.select` | The text shown in the `mode` element for select mode | `"SEL"` |

The following statusline elements can be configured as follows:

<!-- the other existing table -->
```

Reply:
Alright, I updated it to match your suggestion